CONTRIBUTING.md

```diff
@@ -170,6 +170,21 @@ for coding advice). The code must adhere to the following robustness principles
 are important for software that runs autonomously and continuously without direct
 interaction with a human (such as this Collector).
 
+### Naming convention
+
+To keep naming patterns consistent across the project, naming patterns are enforced to make intent clear by:
+
+- Methods that return a variable that uses the zero value or values provided via the method MUST have the prefix `New`
+  - For example `func NewKinesisExporter(kpl aws.KinesisProducerLibrary)` allocates a variable that uses
+    the variables passed on creation.
+  - For example `func NewKeyValueBuilder()` may allocate internal varialbes to a safe zero value
+- Methods that return a variable that uses non zero value(s) that impacts business logic must use `NewDefault`
+  - For example `func NewDefaultKinesisConfig()` would return a configuration that is the suggested default
+    and can be updated without concern of a race condition.
+- Methods that act upon an input variable should have a signature that reflect concisely the logic being done
+- Variable assigned in a package's global scope that is preconfigured with a recommended set of values must use `Default` as the prefix
```

Comment:
These two are not clear to me. Show examples maybe?

Reply:
Let me know if the added examples help clarify this a little more @tigrannajaryan  :)